internal/failure_converter.go

```diff
@@ -91,9 +95,10 @@ func (dfc *DefaultFailureConverter) ErrorToFailure(err error) *failurepb.Failure
 	switch err := err.(type) {
 	case *ApplicationError:
 		failureInfo := &failurepb.ApplicationFailureInfo{
-			Type:         err.errType,
-			NonRetryable: err.nonRetryable,
-			Details:      convertErrDetailsToPayloads(err.details, dfc.dataConverter),
+			Type:           err.errType,
+			NonRetryable:   err.NonRetryable(),
+			Details:        convertErrDetailsToPayloads(err.details, dfc.dataConverter),
+			NextRetryDelay: durationpb.New(err.nextRetryDelay),
```

Comment:
Sorry I missed this before. We should leave this as `nil` if `nextRetryDelay` is `0`.